pkgs/misc/lollypop-portal/default.nix

```diff
@@ -0,0 +1,62 @@
+{ stdenv, fetchFromGitLab, meson, ninja, pkgconfig
+, python36Packages, gnome3, gst_all_1, gtk3, libnotify
+, kid3, easytag, gobjectIntrospection, wrapGAppsHook }:
+
+stdenv.mkDerivation rec {
+  name = "lollypop-portal";
+  version = "0.9.7";
+
+  src = fetchFromGitLab {
+     domain = "gitlab.gnome.org";
+     owner = "gnumdk";
+     repo = name;
+     rev = version;
+     sha256 = "0rn5xmh6391i9l69y613pjad3pzdilskr2xjfcir4vpk8wprvph3";
+  };
+
+  nativeBuildInputs = [ 
+    gobjectIntrospection
+    meson
+    ninja
+    pkgconfig
+    python36Packages.wrapPython
+    wrapGAppsHook
+  ];
+
+  buildInputs = [
+    gnome3.totem-pl-parser
+    gnome3.libsecret
+    gnome3.gnome-settings-daemon
+
+    gst_all_1.gstreamer
+    gst_all_1.gst-plugins-base
+
+    gtk3
+    libnotify
+  ];
+
+  pythonPath = with python36Packages; [
+    pygobject3
+    pydbus
+    pycairo
+  ];
+
+  dontWrapGApps = true;
+
+  postFixup = ''
+    buildPythonPath "$out/libexec/lollypop-portal $pythonPath"
+
+    wrapProgram $out/libexec/lollypop-portal \
```

Comment:
Alternately, you can append to `gappsWrapperArgs`: https://github.com/NixOS/nixpkgs/blob/9bbb84f622e8c421419e6ca9949a5b55e25c4be8/pkgs/desktops/gnome-3/core/nautilus/default.nix#L26-L33

Reply:
Lol That's what I was trying to do

Reply:
It seems that this won't make a wrapper for the program at libexec which is why I think I did this.

Reply:
Works for me:

```diff
--- a/pkgs/misc/lollypop-portal/default.nix
+++ b/pkgs/misc/lollypop-portal/default.nix
@@ -14,7 +14,7 @@
      sha256 = "0rn5xmh6391i9l69y613pjad3pzdilskr2xjfcir4vpk8wprvph3";
   };
 
-  nativeBuildInputs = [ 
+  nativeBuildInputs = [
     gobjectIntrospection
     meson
     ninja
@@ -41,15 +41,12 @@
     pycairo
   ];
 
-  dontWrapGApps = true;
-
-  postFixup = ''
+  preFixup = ''
     buildPythonPath "$out/libexec/lollypop-portal $pythonPath"
-
-    wrapProgram $out/libexec/lollypop-portal \
-      "''${gappsWrapperArgs[@]}" \
-        --prefix PYTHONPATH : "$program_PYTHONPATH" \
-        --prefix PATH : "${stdenv.lib.makeBinPath [ easytag kid3 ]}"
+    gappsWrapperArgs+=(
+      --prefix PYTHONPATH : "$program_PYTHONPATH"
+      --prefix PATH : "${stdenv.lib.makeBinPath [ easytag kid3 ]}"
+    )
   '';
 
   meta = with stdenv.lib; {
```

Reply:
Weird I ended up with the same change and it fails?

Reply:
Yep, produces the same wrapper file.